### PR TITLE
Callbacks improvements

### PR DESCRIPTION
**Pull request by sliwy**
Improves callbacks definiton, so now a user don't need to specify all callbacks using skorch. It is enough to pass scoring functions names from sklearn and `EEGClassifier` will create callbacks internally, as described in #48 
I also added `EEGClassifier` to be displayed in our docs. Docstring needs improvement definitely, (I will create an issue) as for now it does not show docstring of base class.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -20,6 +20,18 @@ This is the reference for classes (``CamelCase`` names) and functions
    :no-members:
    :no-inherited-members:
 
+Classifier
+======
```
Comment (agramfort) [inline docs/api.rst]:
```suggestion
==========
```

we should see if CIs can fail if there are sphinx formatting errors

**Comment by agramfort** [comment]
are all the examples in https://288-232335424-gh.circle-artifacts.com/0/dev/auto_examples/index.html still relevant?
It would be nice to have matplotlib plot something so the thumbnails are appealing in this page.

File: braindecode/classifier.py [hunk 2]
```diff
@@ -1,16 +1,83 @@
...
+        super().__init__(*args, callbacks=callbacks, **kwargs)
+
+    def _parse_callbacks(self, callbacks):
+        callbacks_list = []
+        if callbacks is not None:
+            for callback in callbacks:
+                if isinstance(callback, tuple):
+                    callbacks_list.append(callback)
+                else:
+                    assert isinstance(callback, str)
```
Comment (robintibor) [inline braindecode/classifier.py]: Can you change to 
```suggestion
                    assert isinstance(callback, str) or callable(callback)
```
This is also what get_scorer expects from scikit learn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.get_scorer.html

Comment (robintibor) [reply braindecode/classifier.py]: Ah ok I see you might then have to also change line below with _score_func.name

Comment (robintibor) [reply braindecode/classifier.py]: Hm Ok I see you use assumption you have a string quite often below... Then ignore my comment, maybe for a future time I leave it here

File: braindecode/scoring.py [hunk 3]
```diff
@@ -124,9 +124,7 @@ def __init__(
         name=None,
         target_extractor=to_numpy,
         use_caching=True,
-        input_time_length=None,
```
Comment (robintibor) [inline braindecode/scoring.py]: ah great catch

**Comment by robintibor** [comment]
> are all the examples in https://288-232335424-gh.circle-artifacts.com/0/dev/auto_examples/index.html still relevant?
> It would be nice to have matplotlib plot something so the thumbnails are appealing in this page.

Some will go, as written in #90 
But ya then maybe worthwhile to plot something when cleaning up the example files

**Pull request merged**